lib/more_core_extensions/core_ext/string/formats.rb

```diff
@@ -13,6 +13,12 @@ def domain_name?
       !!(self =~ RE_DOMAINNAME)
     end
 
+    # Based on RFC 1034
+    RE_HOSTNAME = /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])$/
+    def hostname?
```

Comment:
What is the difference between a `hostname` and a `domain_name`?

Reply:
Well, basically the `domain_name?` validation returns `false` for a single string like "localhost" for example, which is an acceptable value for the hostnames fields. And I'm not comfortable enough to update the domain validation, once it could be used on anothers places where this validation make sense.